sys/posix/pthread/pthread.c

```diff
@@ -134,6 +138,11 @@ int pthread_create(pthread_t *newthread, const pthread_attr_t *attr, void *(*sta
     bool autofree = attr == NULL || attr->ss_sp == NULL || attr->ss_size == 0;
     size_t stack_size = attr && attr->ss_size > 0 ? attr->ss_size : PTHREAD_STACKSIZE;
     void *stack = autofree ? malloc(stack_size) : attr->ss_sp;
+
+    if (stack == NULL) {
+        return -ENOMEM;
```

Comment:
Without having looked closely in the code it seems that a `free(pt);` should be added if the second allocation fails to not leak memory.